src/commentscan.l

```diff
@@ -3821,29 +3822,55 @@ static inline void setOutput(yyscan_t yyscanner,OutputContext ctx)
       yyextra->pOutputString = &yyextra->current->doc;
       break;
     case OutputBrief:
-      if (oldContext!=yyextra->inContext)
       {
-        if (yyextra->current->brief.isEmpty()) yyextra->current->briefLine = yyextra->lineNr;
-        if (yyextra->current->briefFile.isEmpty())
+        if (oldContext!=yyextra->inContext)
         {
-          yyextra->current->briefFile = yyextra->fileName;
-          yyextra->current->briefLine = yyextra->lineNr;
+          if (yyextra->current->brief.isEmpty()) yyextra->current->briefLine = yyextra->lineNr;
+          if (yyextra->current->briefFile.isEmpty())
+          {
+            yyextra->current->briefFile = yyextra->fileName;
+            yyextra->current->briefLine = yyextra->lineNr;
+          }
         }
-      }
-      if (yyextra->current->brief.stripWhiteSpace().isEmpty()) // we only want one brief
-        // description even if multiple
-        // are given...
-      {
-        yyextra->pOutputString = &yyextra->current->brief;
-      }
-      else
-      {
-        if (!yyextra->current->doc.isEmpty()) // when appending parts add a new line
+        bool foundMatch = false;
+        if (yyextra->current->brief.stripWhiteSpace().isEmpty()) // we only want one brief
+          // description even if multiple
+          // are given...
+        {
+          foundMatch = true;
+        }
+        else
+        {
+          static const reg::Ex nonBrief(R"( *[\\@]ifile *\"[^\"]*\" *[\\@]ilinebr *[\\@]iline *(\d+) *[\\@]ilinebr *\n)");
+          static const reg::Ex nonBrief1(R"( *[\\@]ifile *\"[^\"]*\" *[\\@]ilinebr *[\\@]iline *\d+ *[\\@]ilinebr *)");
```

Comment:
Looks like this is a longer match, but it is checked below later in the `else` case, so I wonder if there is ever a possibility `nonBrief` is set to `false` while `nonBrief1` is `true`? if not then this can be optimized away.

Reply:
I added the `nonBrief1` as safety measure. The `nonBrief` (i.e. the one with the `\n` ending`) normally is used.

Reply:
I think the 'safety' case can never be reached, i.e. it is dead code.

Reply:
Looks like I missed the extra `\n` for `nonBrief`. Still, I'm not very happy with the additional regex.

Reply:
I agree that the extra regex is not nice.
I didn't find, quickly, the place where the `\n` is added, maybe the end of the regex should read `(\n)?` and later on we test for the `\n`

But maybe something like:
```
/**
 @page  ExamplePage
 This is the example page. The second line.

 */
```
which gives in 1.9.8:
![image](https://github.com/doxygen/doxygen/assets/5223533/3742c44e-e38c-4e21-a317-427204093c8a)

and without the fix:
![image](https://github.com/doxygen/doxygen/assets/5223533/cabee3c0-12fb-4238-af1d-3792400ddc12)

and with the fix this doesn't change, looks like we don't even enter the tests with the regex.

The later case I have to check.
- Should this check also be done in the light of this issue?
- should we change the regex to a version with `(\n)?` and rets for the `\n` already?